src/target/adiv5.c

```diff
@@ -281,6 +286,10 @@ static void adiv5_component_probe(ADIv5_AP_t *ap, uint32_t addr)
 	if (cid_class == cidc_romtab) { /* ROM table, probe recursively */
 		for (int i = 0; i < 256; i++) {
 			uint32_t entry = adiv5_mem_read32(ap, addr + i*4);
+			if (adiv5_dp_error(ap->dp)) {
+				DEBUG("Fault reading ROM table entry\n");
```

Comment:
I don't really think these should be reported as errors.  Failure to read the ROM table is not significantly different from successfully reading the ROM table and not seeing anything we recognise. Both cases just indicate that this isn't a usable target.

In the current source tree this is only ever called from a GDB monitor command, so it's technically possible to send output with `O` GDB packets.  It is, however, conceivable that this target code could be used in another context where this is not available, so I want to avoid putting GDB dependencies into the target code.

`tc_printf` is the clean function for output in monitor commands from the target code, but that can't be used here as there is no target or attached target controller in this context.